HungryDevelopers.playground/Contents.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+
+class Spoon {
+    var index: Int = 0
+    private let lock = NSLock()
+    
+    func pickUp() {
+        lock.lock()
+    }
+    
+    func putDown() {
+        lock.unlock()
+    }
+}
+
+class Developer {
+    var name: String?
+    var leftSpoon: Spoon?
+    var rightSpoon: Spoon?
+    
+    func think() {
+        let group = DispatchGroup()
+        let avenueQueue = DispatchQueue(label: "Avenue Queue")
+        
+        if let leftIndex = leftSpoon?.index,
+            let rightIndex = rightSpoon?.index {
```

Comment:
This also makes the code less clear. I was wondering by looking at this if they were switching spoons or what was going on with the optionals.